### PR TITLE
Add Log Output when DevTools restart is disabled

### PR DESCRIPTION
**Pull request by eiselems**
Also did a small refactoring to get rid of the unused getter and inline
the highest precedence reference.

See gh-14492

### CODE REVIEW

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java [hunk 1]
```diff
@@ -35,7 +38,7 @@
 public class RestartApplicationListener
 		implements ApplicationListener<ApplicationEvent>, Ordered {
 
-	private int order = HIGHEST_PRECEDENCE;
+	private static final Log logger = LogFactory.getLog(RestartApplicationListener.class);
```
Comment (snicoll) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java]: Please refrain from doing unrelated changes. This change to `order`  is unrelated to the task at hand.

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java [hunk 2]
```diff
@@ -81,15 +85,6 @@ private void onApplicationFailedEvent(ApplicationFailedEvent event) {
...
-		return this.order;
-	}
-
-	/**
-	 * Set the order of the listener.
-	 * @param order the order of the listener
-	 */
-	public void setOrder(int order) {
-		this.order = order;
+		return HIGHEST_PRECEDENCE;
```
Comment (snicoll) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java]: I have no opinion has whether this should be changed but we should have this discussion in a separate PR.

Comment (eiselems) [reply spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java]: changed it back

File: spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java [hunk 3]
```diff
@@ -67,6 +72,7 @@ private void onApplicationStartingEvent(ApplicationStartingEvent event) {
 			Restarter.initialize(args, false, initializer, restartOnInitialize);
 		}
 		else {
+			logger.info("RestartApplicationListener is disabled.");
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java]: Logging here could be useful, particularly if it indicated when the restarter hasn't been initialised, but it doesn't meet the requirement described in #14492. That issue is specifically interested in the case where `restartOnInitialize` is false because an agent reloader is active.

Comment (eiselems) [reply spring-boot-project/spring-boot-devtools/src/main/java/org/springframework/boot/devtools/restart/RestartApplicationListener.java]: Mhm maybe I gonna close this one and bring it together if I find a resolution to satisfy #14492 then.
